client/blocks/plan-storage/bar.jsx

```diff
@@ -14,19 +13,10 @@ export class PlanStorageBar extends Component {
 		className: PropTypes.string,
 		mediaStorage: PropTypes.object,
 		displayUpgradeLink: PropTypes.bool,
-		sitePlanSlug: PropTypes.string.isRequired,
 	};
 
 	render() {
-		const { className, displayUpgradeLink, mediaStorage, sitePlanSlug, translate } = this.props;
-
-		if ( planHasFeature( sitePlanSlug, FEATURE_UNLIMITED_STORAGE ) ) {
-			return null;
-		}
-
-		if ( ! mediaStorage || mediaStorage.maxStorageBytes === -1 ) {
-			return null;
-		}
```

Comment:
I've moved this check to the parent component as well, so we can avoid rendering it all on the `PlanCard` when there is no storage bar to render.